solr/core/src/java/org/apache/solr/servlet/HttpSolrCall.java

```diff
@@ -1170,6 +1171,21 @@ protected Object _getHandler() {
     return handler;
   }
 
+  /** Gets the client (user-agent) SolrJ version, or null if isn't SolrJ. */
+  public SolrVersion getUserAgentSolrVersion() {
+    String header = req.getHeader("User-Agent");
+    if (header == null || !header.startsWith("Solr")) {
+      return null;
+    }
+    try {
+      return SolrVersion.valueOf(header.substring(header.lastIndexOf(' ') + 1));
```

Comment:
If this gets an older client it will return 1.0 or 2.0, which can be somewhat misleading.

Reply:
Yeah; I thought it would have no consequence so I did nothing about it.  At least by returning a SolrVersion instead of null, we communicate indirectly that it's at least SolrJ.  I could add a javadoc comment on 1.0 and 2.0 being used prior to 9.8.

Reply:
I added this to the docs.

Reply:
BTW that method has no callers; isn't tested.  I did manually test it.  Obviously I intend to have it be used soon.